props/saber_fett263_buttons.h

```diff
@@ -381,160 +1755,2779 @@ SaberFett263Buttons() : PropBase() {}
+              return true;
+          }
+          STDOUT.println("Enter Edit Mode");
+          GenerateIniFiles();
+          PlayMenuSound("medit.wav");
+          current_menu_angle_ = fusor.angle2();
+          menu_top_pos_ = 0;
+          menu_ = true;
+          menu_type_ = MENU_TOP;
+          FastOn();
+        } 
+#else
+        GenerateIniFiles();
+        PlayMenuSound("msave.wav");
+#endif
+        return true;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_RIGHT, MODE_OFF | BUTTON_POWER):
+        if (menu_ && menu_type_ == TRACK_PLAYER) {
+          track_mode_ = PLAYBACK_ROTATE;
+          PlayMenuSound("mrotate.wav");
+          return true;
+        }
+        return false;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_LEFT, MODE_OFF | BUTTON_POWER):
+        if (menu_ && menu_type_ == TRACK_PLAYER) {
+          track_mode_ = PLAYBACK_LOOP;
+          PlayMenuSound("mloop.wav");
+          return true;
+        }
+        return false;
+
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_ON):
+        if (menu_) {
+          PlayMenuSound("mexit.wav");
+          menu_ = false;
+          menu_type_ = MENU_TOP;
+          return true;
+        }
+        if (rehearse_) {
+          PlayMenuSound("mcancel.wav");
+          rehearse_ = false;
+          return true;
+        }
+        if (!battle_mode_) {
+          battle_mode_ = true;
+          if (SFX_bmbegin) {
+            hybrid_font.PlayCommon(&SFX_bmbegin);
+          } else {
+            hybrid_font.DoEffect(EFFECT_FORCE, 0);
+          }
+        } else {
+          battle_mode_ = false;
+          if (SFX_bmend) {
+            hybrid_font.PlayCommon(&SFX_bmend);
+          } else {
+            beeper.Beep(0.5, 3000);
+          }
+        }
+        return true;
+
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_ON | BUTTON_POWER):
+        if (!menu_ && saved_gesture_control.powerlock || choreo_) {
+          wav_player.Free();
+          choreo_ = false;
+          battle_mode_ = false;
+          Off();
+        }
+        return true;
+
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_RIGHT, MODE_ON):
+        if (wav_player->isPlaying()) return true;
+        if (color_mode_ == CC_COLOR_LIST) {
+          dial_ = (dial_ + 1) % NELEM(color_list_);
+          ONCEPERBLADE(CC_NEW_COLOR)
+          current_preset_.Save();
+          hybrid_font.PlayCommon(&SFX_ccchange);
+          UpdateStyle(current_preset_.preset_num);   
+          return true;
+        }
+        if (!menu_) return false;
+        switch (menu_type_) {
+          case MENU_PRESET:
+            next_preset_fast();
+            return true;
+          case MENU_VOLUME:
+             VolumeUp();
+             return true;
+#ifdef FETT263_EDIT_MODE_MENU
+          case MENU_TOP:
+            menu_top_pos_ = (menu_top_pos_ + 1) % SUBMENUS;
+            if (menu_top_pos_ == 0) menu_top_pos_ = 1;
+            switch (menu_top_pos_) {
+              case 1:
```

Comment:
enum?

Reply:
the submenus vary too much (they are reused in several places) to have an enum that would make any sense so I just used a position (step) to determine